cpp/kiss_icp/core/Registration.hpp

```diff
@@ -41,6 +43,8 @@ struct Registration {
 
     int max_num_iterations_;
     double convergence_criterion_;
-    int max_num_threads_;
+    // This attribute requires static to be able to manipulate the max
+    // concurrency from TBB across multiple instances of this class
+    static tbb::global_control tbb_control_settings;
```

Comment:
@nachovizzo do you mean something like this?

Reply:
I don't think the default constructor is really needed from our side, I think I wanted mostly to make it right, although a bit uglier. I think we can vote and decide altogether because this is not critical. But let's make it correct more so that beautiful but hidden.

Reply:
All good, then if this is the target of the PR we should rename the PR or close/reopen. Otherwise, It's extremely confusing for me that open this at different intervals.

I'm happy if there is quorum to remove the static hack, but let's vote

Reply:
> We can't know the max range of your sensor at compile time, so we "force" you to specify this dynamically.

Well, this we don't really enforce because we have default configs in both the pipeline header and the ROS launch file. The main idea of this PR was more: If you provide a default config, you can also provide a default constructor.

If we agree not to allow a default constructor, we should be more consistent and also not provide a default config. Otherwise, people can anyway just initialize with the default config.

Reply:
You are right, we are not enforcing per-se. In fact, I don't believe anyone actually realizes that we have some parameters to tune 🤔 which is a success I'd say.

The main difference for the C++ side is that it's more likely users will indeed instantiate an object of the pipeline class, in contrast to the python case where we don't really expect that use case and people will use it more as a black-box system. In fact, one can't bake a `KissICP` object in python without providing a config
<img width="914" alt="image" src="https://github.com/PRBonn/kiss-icp/assets/21349875/f160f002-21ec-47ac-917c-7971a5f56762">

If I'm writing C++ code, I'd love to see this error at compile time, and at least be conisuous of constructing the object with a default config, which I don't even want to look into maybe, but understand that there is a config.
In either case, it's a matter of taste and decision

Reply:
Okay, that's a fair point.

Regarding making the tbb control setting an attribute I don't have a strong opinion, so I am fine with either renaming this PR or closing it.

Reply:
It seems none of us strongly need/want/desire the tbb control as an attribute or the default constructor. Let's close this PR for now and reevaluate later if something changes. thanks, guys